thunder/distributed/__init__.py

```diff
@@ -38,6 +38,43 @@
 _skip_data_parallel_grad_sync = ContextVar("skip_data_parallel_grad_sync", default=False)
 
 
+def _avoid_torch_nccl_record_streams(func):
+    """
+    Avoids the allocator thrashing issue in PyTorch NCCL backend.
+    """
+
+    env_var = "TORCH_NCCL_AVOID_RECORD_STREAMS"
+    value = os.environ.get(env_var, "0")
+
+    def wrapper(*args, **kwargs):
+        try:
+            os.environ[env_var] = "1"
+            return func(*args, **kwargs)
+        finally:
+            os.environ[env_var] = value
+
+    return wrapper
+
+
+@_avoid_torch_nccl_record_streams
+def copy_default_process_group() -> ProcessGroup:
+    """Create a new process group with the same ranks as the default process group.
+
+    Returns:
+        A new process group with the same ranks as the default process group.
+    """
+    default_pg = tdist.distributed_c10d._get_default_group()
+    ranks = list(range(tdist.get_world_size(group=default_pg)))
+    backend = tdist.distributed_c10d.get_backend(default_pg)
+    # What's the better way to query this from the default process group? This
+    # is the default value for `is_high_priority_stream` in PyTorch
+    # default_pg.options returns ProcessGroup.Options object while
+    # ProcessGroupNCCL.Options is required
+    options = tdist.ProcessGroupNCCL.Options()
```

Comment:
We do test ddp on CPU, so this would need to be guarded on the backend being NCCL.

Reply:
```
FAILED thunder/tests/distributed/test_ddp.py::test_native_ddp_torch_cpu_float32[0] - AttributeError: module 'torch.distributed' has no attribute 'ProcessGroupNCCL'
FAILED thunder/tests/distributed/test_ddp.py::test_native_ddp_torch_cpu_float32[25] - AttributeError: module 'torch.distributed' has no attribute 'ProcessGroupNCCL'
``